File: back_end/src/main/java/pt/iade/RunUp/model/dto/RouteRequest.java
package pt.iade.RunUp.model.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RouteRequest {
    private String nome;
    private double originLat;
    private double originLng;
    private double destLat;
    private double destLng;
    private double desiredDistanceKm;
    private boolean preferTrees;
    private boolean nearBeach;
    private boolean nearPark;
    private boolean sunnyRoute;
    private boolean avoidHills;
    private String tipo; 
}

File: front_end/app/src/main/java/pt/iade/ei/runupsetup/models/RouteRequest.kt
package pt.iade.ei.runupsetup.models

import java.io.Serializable

data class RouteRequest(
    val nome: String,
    val originLat: Double,
    val originLng: Double,
    val destLat: Double,
    val destLng: Double,
    val desiredDistanceKm: Double,
    val preferTrees: Boolean,
    val nearBeach: Boolean,
    val nearPark: Boolean,
    val sunnyRoute: Boolean,
    val avoidHills: Boolean,
    val tipo: String
) : Serializable
